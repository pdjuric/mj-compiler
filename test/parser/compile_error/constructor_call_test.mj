program constructor_call_test

class A {
    {
        A(int a, char c) {}
    }
}

{
    void main() A a1, a2, a3; {
        a1 = new int();
        a2 = new a1();

        a1 = new A();
        a1 = new A(5, 'c');
        a1 = new A(5);
        a1 = new A(5, 'c', true);
        
    }


}