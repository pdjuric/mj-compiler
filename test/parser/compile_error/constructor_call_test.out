INFO  16:01:19,319 - line  20 - Access to local variable a1 [Var a1: Class A, 0, 1 ].
ERROR 16:01:19,320 - line  20 - int is not a class
INFO  16:01:19,320 - line  20 - Call to constructor of class [unknown class] [[]].
INFO  16:01:19,321 - line  21 - Access to local variable a2 [Var a2: Class A, 1, 1 ].
ERROR 16:01:19,321 - line  21 - a1 is not a type
ERROR 16:01:19,322 - line  21 - a1 is not a class
INFO  16:01:19,322 - line  21 - Call to constructor of class [unknown class] [[]].
INFO  16:01:19,322 - line  23 - Access to local variable a1 [Var a1: Class A, 0, 1 ].
ERROR 16:01:19,331 - line   0 - No matching constructor of class A for provided arguments.
INFO  16:01:19,331 - line  23 - Call to constructor of class [unknown class] [[]].
INFO  16:01:19,331 - line  24 - Access to local variable a1 [Var a1: Class A, 0, 1 ].
INFO  16:01:19,332 - line  24 - Call to constructor of class A [Meth __ctor_0: , -1, 3 
   Var this: , 0, 2 
   Var a: int, 1, 2 
   Var c: char, 2, 2 
].
INFO  16:01:19,332 - line  25 - Access to local variable a1 [Var a1: Class A, 0, 1 ].
ERROR 16:01:19,333 - line  25 - No matching constructor of class A for provided arguments.
INFO  16:01:19,333 - line  25 - Call to constructor of class [unknown class] [[]].
INFO  16:01:19,333 - line  26 - Access to local variable a1 [Var a1: Class A, 0, 1 ].
ERROR 16:01:19,334 - line  26 - No matching constructor of class A for provided arguments.
INFO  16:01:19,334 - line  26 - Call to constructor of class [unknown class] [[]].